### PR TITLE
Remove accessors for deprecated status code, fix receiver logic

### PR DESCRIPTION
**Pull request by bogdandrutu**
Updates #2488

The previous logic was to ignore deprecated if received non unset for new status code,
but if downstream a service is not upgraded it should see the deprecated status set correctly.

This change is to be consistent with `SetCode`.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/otlpreceiver/trace/otlp.go [hunk 1]
```diff
@@ -72,9 +72,16 @@ func (r *Receiver) Export(ctx context.Context, req *collectortrace.ExportTraceSe
...
+				switch span.Status.Code {
+				case otlptrace.Status_STATUS_CODE_UNSET:
+					if span.Status.DeprecatedCode != otlptrace.Status_DEPRECATED_STATUS_CODE_OK {
+						span.Status.Code = otlptrace.Status_STATUS_CODE_ERROR
+					}
+				case otlptrace.Status_STATUS_CODE_OK:
+					// If status code is set then overwrites deprecated.
+					span.Status.DeprecatedCode = otlptrace.Status_DEPRECATED_STATUS_CODE_OK
+				case otlptrace.Status_STATUS_CODE_ERROR:
+					span.Status.DeprecatedCode = otlptrace.Status_DEPRECATED_STATUS_CODE_UNKNOWN_ERROR
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/trace/otlp.go:84]: Why these changes? The spec says that receivers MUST:

> //   If code!=STATUS_CODE_UNSET then the value of `deprecated_code` MUST be
>   //   ignored, the `code` field is the sole carrier of the status.

This does not seem to match the spec anymore.

Comment (bogdandrutu) [reply receiver/otlpreceiver/trace/otlp.go:84]: What does the spec says? Why does "SetCode" updates deprecated but in this case we don't?

Comment (bogdandrutu) [reply receiver/otlpreceiver/trace/otlp.go:73]: Updated comments to SetCode as well as for the receiver to clarify when we are sender vs receiver.

Comment (tigrannajaryan) [reply receiver/otlpreceiver/trace/otlp.go:73]: Discussed offline, so yes, OTLP receiver is both a "receiver" and "sender" in the sense that the proto uses these words here: https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L239

**Review by tigrannajaryan — APPROVED** [review]
LGTM
